practice/src/javascripts/sidebar.js

```diff
@@ -2,12 +2,17 @@
 document.querySelector('.nav-list').addEventListener('click', function (event) {
   // Check if the clicked element is a nav-link inside a nav-item
   if (event.target.classList.contains('nav-link')) {
-    // Remove 'active' class from all nav-items
-    document
-      .querySelectorAll('.nav-item.active')
-      .forEach((item) => item.classList.remove('active'));
+    const clickedNavItem = event.target.closest('.nav-item');
 
-    // Add 'active' class to the parent nav-item of the clicked nav-link
-    event.target.closest('.nav-item').classList.add('active');
+    // Check if the clicked nav-item is already active
+    if (!clickedNavItem.classList.contains('active')) {
+      // Remove 'active' class from all nav-items
+      document
+        .querySelectorAll('.nav-item.active')
```

Comment:
@JennyDoJD why should we need `querySelectorAll` if only one active item as same time?